test/Test.hs

```diff
@@ -0,0 +1,33 @@
+module Main where
+
+import Data.Foldable (for_)
+import System.Directory (listDirectory, removeDirectoryRecursive, removeFile)
+import System.Exit (exitFailure)
+import System.FilePath (isExtensionOf, splitExtension, (<.>), (</>))
+
+main :: IO ()
+main = do
+    files <- map ("test/Test" </>) <$> listDirectory "test/Test/"
+    let testFiles = filter ("test" `isExtensionOf`) files
+    -- compare outputs of .test and .golden files
+    result <- traverse compareWithGolden testFiles >>= pure . foldr (&&) True
+
+
+    -- clean up all .test files
+    putStrLn ("Cleaning tests" :: String)
```

Comment:
If `-XOverloadedStrings` extension is not enabled and if you don't use `relude` then having `:: String` is not necessary. But with `Relude` you can use `putTextLn`.